### PR TITLE
Add CORS

### PR DESCRIPTION
**Pull request by joshsmith**
This adds CORS using `corsica` to be sure we can actually make requests to the server with our Ember application.

Testing inside the PageController test.

### CODE REVIEW

**Comment by joshsmith** [comment]
If anyone has a good idea for how to write a test for this, feel free to run with it. This is blocked until that time.

_EDIT_: now testing inside the PageController test.

**Comment by joshsmith** [comment]
This could use a review from anyone now.

File: lib/code_corps/endpoint.ex [hunk 1]
```diff
@@ -38,5 +38,6 @@ defmodule CodeCorps.Endpoint do
     key: "_code_corps_key",
     signing_salt: "RJebPcfU"
 
+  plug Corsica, origins: "http://localhost:4200"
```
Comment (noslouch) [inline lib/code_corps/endpoint.ex]: The allowed origins should probably be defined in the env config files and hen accessed here. I think something like:

``` elixir
config :code_corps, allowed_origins: {"http://localhost:4200"} 
```

in config/dev.ex and then accessed here like

``` elixir
Application.get_env :code_corps, :allowed_origins
```

would do the trick

File: test/controllers/page_controller_test.exs [hunk 2]
```diff
@@ -5,4 +5,20 @@ defmodule CodeCorps.PageControllerTest do
...
+    assert "http://localhost:4200" in get_resp_header(conn, "access-control-allow-origin")
+  end
+
+  test "does not allow CORS for http://google.com" do
+    conn = build_conn()
+      |> put_req_header("origin", "http://google.com")
+      |> get("/")
+
+    refute "http://google.com" in get_resp_header(conn, "access-control-allow-origin")
+  end
```
Comment (noslouch) [inline test/controllers/page_controller_test.exs:23]: Should this also test for pre-flight requests?

Comment (joshsmith) [inline test/controllers/page_controller_test.exs:23]: I don't know if we need to do more than a smoke test to see if the plug is there. What do you think?

File: lib/code_corps/endpoint.ex (hunk 1 above)
Comment (joshsmith) [inline lib/code_corps/endpoint.ex]: Good call. Any idea if we should be testing all the envs then or not?

File: test/controllers/page_controller_test.exs (hunk 2 above)
Comment (noslouch) [inline test/controllers/page_controller_test.exs:23]: Fair enough. I guess this is mostly testing that we've got CORS enabled and Corsica can do the rest. 

Though I wonder if there's value is faking different envs to test the allowed origins? Might be overkill.

Comment (joshsmith) [inline test/controllers/page_controller_test.exs:23]: Yeah that's the way I'd lean, against the overkill. Churn is low on this anyway. Just wanted some basic testing.

File: lib/code_corps/endpoint.ex (hunk 1 above)
Comment (noslouch) [inline lib/code_corps/endpoint.ex]: Ha see my other comment. Might be overkill to update a config and then update a test for that config. But then again when a new subdomain is added and it breaks the app we'll know who to blame.

**Pull request merged**